### PR TITLE
Blip dynamic input resolution

### PR DESCRIPTION
**Pull request by zafstojano**
# What does this PR do?

This PR adds ability to handle dynamic input image resolutions to the BLIP family of models.

Addresses #30579 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

Ping: @amyeroberts

### CODE REVIEW

File: tests/models/blip/test_modeling_blip.py [hunk 1]
```diff
@@ -1381,6 +1381,17 @@ def test_inference_image_captioning_fp16(self):
...
+    def test_inference_interpolate_pos_encoding(self):
+        model = BlipForConditionalGeneration.from_pretrained("Salesforce/blip-image-captioning-base").to(torch_device)
+        processor = BlipProcessor.from_pretrained("Salesforce/blip-image-captioning-base")
+        processor.image_processor.size = {"height": 500, "width": 500}
+
+        image = prepare_img()
+        inputs = processor(images=image, return_tensors="pt").to(torch_device)
+
+        predictions = model.generate(**inputs, interpolate_pos_encoding=True)
+        self.assertEqual(predictions[0].tolist(), [30522, 1037, 2450, 3564, 2006, 1996, 3509, 2007, 1037, 3899, 102])
```
Comment (amyeroberts) [inline tests/models/blip/test_modeling_blip.py:1393]: Have you checked the generated ids here to make sure they make sense for the picture i.e.:

`processor.batch_decode([30522, 1037, 2450, 3564, 2006, 1996, 3509, 2007, 1037, 3899, 102])`

**Review by amyeroberts — COMMENTED** [review]
Amazing work - thanks for adding this! 

Just a small Q on the test outputs. Once sense-checked we'll be good to merge!

**Comment by zafstojano** [comment]
@amyeroberts good point!

the tests now also have a check for the textual content.

**Review by amyeroberts — APPROVED** [review]
Amazing work - thanks for adding this!

**Pull request merged**